src/codewhisperer/tracker/codewhispererCodeCoverageTracker.ts

```diff
@@ -238,21 +240,53 @@ export class CodeWhispererCodeCoverageTracker {
         this.addTotalTokens(filename, text.length)
     }
 
+    // For below 2 edge cases
+    // 1. newline character with indentation
+    // 2. 2 character insertion of closing brackets
+    public getCharacterCountFromComplexEvent(e: vscode.TextDocumentChangeEvent) {
+        if (e.document.languageId === 'java' && e.contentChanges.length === 2) {
```

Comment:
why do we need `e.document.languageId === 'java'` ?

Reply:
Because this API behaves differently in Java files.

Reply:
I tested it again. 

In other languages, there is a small chance that `e.contentChanges.length` becomes 2, while one is a empty event, another is a merged event. So I am improving the condition in last commit. This is tricky but I am able to reproduce.